swap/tests/testutils/mod.rs

```diff
@@ -320,12 +330,31 @@ where
         xmr: swap_amounts.xmr * 10,
         btc: bitcoin::Amount::ZERO,
     };
+
+    let port = get_port().expect("Failed to find a free port");
```

Comment:
You could just let the OS assign you a port by passing 0 to the multi-addr. You'll need to wait for the `NewListenAddr` event then but you will probably need to do that anyway to make sure the Swarm is ready :)

Reply:
Ah, so the way to do it is to listen to `NewListenAddr` - we should give this a try @rishflab ! - This might solve the dialing test failure when concurrently executing tests.

Reply:
https://github.com/comit-network/xmr-btc-swap/issues/151